config/configparser/config.go

```diff
@@ -150,70 +145,31 @@ func Load(v *config.Parser, factories component.Factories) (*config.Config, erro
 	return &cfg, nil
 }
 
-// DecodeTypeAndName decodes a key in type[/name] format into type and fullName.
-// fullName is the key normalized such that type and name components have spaces trimmed.
-// The "type" part must be present, the forward slash and "name" are optional. typeStr
-// will be non-empty if err is nil.
-func DecodeTypeAndName(key string) (typeStr config.Type, fullName string, err error) {
-	items := strings.SplitN(key, typeAndNameSeparator, 2)
-
-	if len(items) >= 1 {
-		typeStr = config.Type(strings.TrimSpace(items[0]))
-	}
-
-	if len(items) == 0 || typeStr == "" {
-		err = errors.New("type/name key must have the type part")
-		return
-	}
-
-	var nameSuffix string
-	if len(items) > 1 {
-		// "name" part is present.
-		nameSuffix = strings.TrimSpace(items[1])
-		if nameSuffix == "" {
-			err = errors.New("name part must be specified after " + typeAndNameSeparator + " in type/name key")
-			return
-		}
-	} else {
-		nameSuffix = ""
-	}
-
-	// Create normalized fullName.
-	if nameSuffix == "" {
-		fullName = string(typeStr)
-	} else {
-		fullName = string(typeStr) + typeAndNameSeparator + nameSuffix
-	}
-
-	err = nil
-	return
-}
-
 func errorInvalidTypeAndNameKey(component, key string, err error) error {
 	return &configError{
 		code: errInvalidTypeAndNameKey,
 		msg:  fmt.Sprintf("invalid %s type and name key %q: %v", component, key, err),
 	}
 }
 
-func errorUnknownType(component string, typeStr config.Type, fullName string) error {
+func errorUnknownType(component string, id config.ID) error {
 	return &configError{
 		code: errUnknownType,
-		msg:  fmt.Sprintf("unknown %s type %q for %s", component, typeStr, fullName),
+		msg:  fmt.Sprintf("unknown %s type %q for %v", component, id.Type(), id),
```

Comment:
We want this to be user readable, so maybe this:
```suggestion
		msg:  fmt.Sprintf("unknown %s type %q for %v", component, id.Type(), id.String()),
```
Similarly for all other errors.

Reply:
The `"%v"` prints the `String` by default so no need to call String, that's why I changed from `"%s"`